frontend/packages/kubevirt-plugin/src/components/vm-templates/vm-template-source.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+import { TemplateKind } from '@console/internal/module/k8s';
+import { Url } from '../../utils/url';
+import { ProvisionSource } from '../../constants/vm/provision-source';
+import { getBasicID, prefixedID } from '../../utils';
+
+const Type: React.FC<TypeProps> = ({ type, source, error, id, isInline = false }) => (
+  <div
+    id={id}
+    title={source || error}
+    className={isInline ? 'kubevirt-template-source__overlay' : ''}
+  >
+    {type}
+  </div>
+);
+
+type TypeProps = {
+  type: string;
+  source?: string;
+  error?: string;
+  id?: string;
+  isInline?: boolean;
+};
+
+const Source: React.FC<SourceProps> = ({ type, source, id }) => {
+  if (!source) {
+    return null;
+  }
+
+  const sourceElem = type === ProvisionSource.URL.getValue() ? <Url url={source} short /> : source;
+
+  return (
+    <div id={id} className="kubevirt-template-source__source">
```

Comment:
Same as above, not sure we need the `id` attribute?

Reply:
done